lib/facter/custom_facts/core/execution/base.rb

```diff
@@ -70,8 +72,7 @@ def execute_command(command, on_fail = nil, logger = nil, time_limit = nil)
             opts = { 'LC_ALL' => 'C', 'LANG' => 'C' }
             require 'timeout'
             @log.debug("Executing command: #{command}")
-            out, stderr = Open3.popen3(opts, command.to_s) do |_, stdout, stderr, wait_thr|
-              pid = wait_thr.pid
+            out, stderr = Popen3.popen3e(opts, command.to_s) do |_, stdout, stderr, pid|
```

Comment:
It would better be something like `Open3.popen3_e` to suggest that is patch of Open3 module with `_e` as it provides the child exit status